keanu-python/tests/test_stats.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import pytest
+
+from itertools import islice
+from keanu import Model, stats, KeanuRandom
+from keanu.algorithm import sample, generate_samples
+from keanu.vertex import Uniform
+import pandas as pd
+
+
+def test_can_get_correct_autocorrelation() -> None:
+    x = [1, 2, 3, 4, 5, 6, 7, 8]
+    x_list = [np.array(a, float) for a in x]
+    autocorr = stats.autocorrelation(x_list)
+    expected = [1., 0.625, 0.27380952, -0.0297619, -0.26190476, -0.39880952, -0.41666667, -0.29166667]
+    np.testing.assert_almost_equal(autocorr, expected)
+
+
+def test_can_get_correct_autocorrelation_when_nonscalar_input() -> None:
+    x_list = [
+        np.array([[1., 27.2], [10., 3.2]]),
+        np.array([[2., 99.4], [5., 4.6]]),
+        np.array([[3., 31.5], [10., 7.8]]),
+        np.array([[4., 14.3], [5., 2.1]]),
+    ]
+    expected = np.array([[[1., 0.25, -0.3, -0.45], [1., -0.27679699, -0.32759603, 0.10439302]],
+                         [[1., -0.75, 0.5, -0.25], [1., -0.40761833, -0.24778339, 0.15540172]]])
+    actual = np.array([[stats.autocorrelation(x_list, (0, 0)),
+                        stats.autocorrelation(x_list, (0, 1))],
+                       [stats.autocorrelation(x_list, (1, 0)),
+                        stats.autocorrelation(x_list, (1, 1))]])
+    np.testing.assert_almost_equal(actual, expected)
+
+
+def test_autocorr_returns_ndarray_of_correct_dtype() -> None:
+    with Model() as m:
+        m.uniform = Uniform(0, 1000)
+    net = m.to_bayes_net()
+    samples = sample(net=net, sample_from=net.get_latent_vertices(), algo="metropolis", draws=1000)
+    valid_key = list(samples.keys())[0]
+    autocorr = stats.autocorrelation(samples.get(valid_key))  # type: ignore
+    assert type(autocorr) == np.ndarray
+    assert autocorr.dtype == samples.get(valid_key)[0].dtype  # type: ignore
```

Comment:
Please add a comment in the code that says why you need to `type: ignore` here

Reply:
Fixed by your solution on other thread.